lib/ansible/modules/packaging/os/redhat_repository.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/python
+
+# Giovanni Sciortino (giovannibattistasciortino@gmail.com)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: redhat_repository
+short_description: Manage RHSM repositories using the C(subscription-manager) command
+description:
+    - Manage(List/Enable/Disable) RHSM repositories to the Red Hat Subscription Management entitlement platform using the C(subscription-manager) command
+version_added: "2.4"
+author: "Giovanni Sciortino (giovannibattistasciortino@gmail.com)"
+notes:
+    - In order to manage rhsm repositories the system must be already registered to rhsm manually or using the ansible module redhat_subscription.
+    - One option between state and list must be defined, both options in the same task must not be defined
+    - When the option state is defined, the option name must be defined
+
+
+requirements:
+    - subscription-manager
+options:
+    state:
+        description:
+          state indicates the desired repositories states.
+        choices: [ "enabled", "disabled" ]
+        required: False
+        default: None
+    name:
+        description:
+          The id of repositories to enable. To operate on several repositories this can accept a comma separated list or a list.
```

Comment:
`id` should be `ID`.

`a comma separated list or a list` should be `a comma separated list or a YAML list`